src/parser/jobs/vpr/modules/Swiftscaled.tsx

```diff
@@ -0,0 +1,31 @@
+import {Event} from 'event'
+import {filter} from 'parser/core/filter'
+import CastTime from 'parser/core/modules/CastTime'
+
+export class Swiftscaled extends CastTime {
+	private SwiftscaledIndex: number | null = null
+
+	override initialise() {
+		super.initialise()
+
+		const fukaFilter = filter<Event>()
+			.target(this.parser.actor.id)
+			.status(this.data.statuses.SWIFTSCALED.id)
+
+		this.addEventHook(fukaFilter.type('statusApply'), this.onApplySwiftscaled)
+		this.addEventHook(fukaFilter.type('statusRemove'), this.onRemoveSwiftscaled)
+	}
+
+	private onApplySwiftscaled(): void {
+		// If this is a reapply for a currently open Fuka window, do not reapply
+		if (this.SwiftscaledIndex == null) {
+			const swiftscaled = this.data.statuses.SWIFTSCALED
+			this.SwiftscaledIndex = this.setPercentageAdjustment('all', swiftscaled.speedModifier, 'both')
```

Comment:
Does Viper have any GCDs locked at 1.5s like Reaper's Enshroud GCDs? I haven't re-checked whether CastTime is smart enough to floor at 1.5s, but I think setting this percentage adjustment for all GCDs might be messing with the ABC percentage, this log is showing over 100% uptime: http://127.0.0.1:3000/fflogs/h2KNPmFfRQxCgJLr/12/298.

Reply:
I will have to test, I think everything runs off swiftscaled, but there are longer and shorter GCD bases. Generations are 2 seconds I think base.

Reply:
Testing complete. Everything is affected by swiftscaled, however there are 5 different GCD speeds that VPR has. It's possible this is causing the over 100%.